build.rs

```diff
@@ -1,191 +1,393 @@
+                                        conf.freeimage_dir));
+        },
+        _ => fail("Invalid freeimage build option set"),
+    };
 
-  if conf.build_cuda == "ON" {
-    backends.push("afcuda".to_string());
-    if !conf.use_lib {
-      backend_dirs.push(build_dir.join("src/backend/cuda").to_str().to_owned().unwrap().to_string());
-    }
-  }
-  if conf.build_opencl == "ON" {
-    backends.push("forge".to_string());
-    backends.push(("afopencl".to_string()));
-    if !conf.use_lib{
-      backend_dirs.push(build_dir.join("third_party/forge/lib").to_str().to_owned().unwrap().to_string());
-      backend_dirs.push(build_dir.join("src/backend/opencl").to_str().to_owned().unwrap().to_string());
-    }
-  }
-  if conf.build_cpu == "ON" {
-    backends.push("afcpu".to_string());
-    if !conf.use_lib{
-      backend_dirs.push(build_dir.join("src/backend/cpu").to_str().to_owned().unwrap().to_string());
-    }
-  }
+    let mut cmake_cmd = Command::new("cmake");
+    cmake_cmd.current_dir(&build_dir);
 
-  if conf.use_lib{
-    backend_dirs.push(conf.lib_dir.to_owned());
-  }
+    run(cmake_cmd.arg("..").arg("-G").arg("Visual Studio 12 2013 Win64")
+        .args(&[format!("-DCMAKE_BUILD_TYPE:STRING={}", conf.build_type),
+                format!("-DBUILD_CPU:BOOL={}", conf.build_cpu),
+                format!("-DBUILD_CUDA:BOOL={}", conf.build_cuda),
+                format!("-DBUILD_OPENCL:BOOL={}", conf.build_opencl),
+                format!("-DBUILD_EXAMPLES:BOOL={}", conf.build_examples),
+                format!("-DBUILD_TEST:BOOL={}", conf.build_test),
+                format!("-DBOOST_ROOT={}", conf.boost_dir),
+                format!("-DCMAKE_INSTALL_PREFIX={}", "package")])
+        .args(&freeimage_options)
+        .args(&fft_options)
+        .args(&blas_options)
+        .args(&lapack_options)
+        .args(&graphics_options)
+        , "cmake");
 
-  return (backends, backend_dirs);
+    let mut make_cmd= Command::new("C:\\Program Files (x86)\\MSBuild\\12.0\\Bin\\MSBuild.exe");
+    make_cmd.current_dir(&build_dir);
+    run(make_cmd
+        .arg(format!("/m:{}", conf.build_threads))
+        .arg(format!("/p:Configuration={}", conf.build_type))
+        .arg(format!("ArrayFire.sln")),
+        "MSBuild");
+    let mut install_cmd= Command::new("C:\\Program Files (x86)\\MSBuild\\12.0\\Bin\\MSBuild.exe");
+    install_cmd.current_dir(&build_dir);
+    run(install_cmd
+        .arg(format!("/p:Configuration={}", conf.build_type))
+        .arg(format!("INSTALL.vcxproj")),
+        "Install");
 }
 
-fn main() {
-  // Setup pathing
-  let src = PathBuf::from(&env::var("CARGO_MANIFEST_DIR").unwrap());
-  let conf_file = src.join("build.conf");
-  let conf = read_conf(&conf_file);
-
-  let mut arrayfire_dir = src.join("arrayfire");
-  let build_dir = arrayfire_dir.join("build");
-  let src_dir = src.join("src");
-  
-  if !conf.use_lib {
+#[cfg(not(windows))]
```

Comment:
We should be able to merge this into one method and use some specific OS checking internally right? Seems excessive.